server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -295,6 +294,15 @@ private void processIndexShardSnapshots(ClusterChangedEvent event) {
                                 }
                             }
                         }
+                    } else {
+                        final ImmutableOpenMap<ShardId, ShardSnapshotStatus> clusterStateShards = entry.shards();
+                        for (ObjectObjectCursor<ShardId, ShardSnapshotStatus> curr : clusterStateShards) {
+                            if (curr.value.state() == State.ABORTED) {
+                                // due to CS batching we might have missed the INIT state and straight went into ABORTED
+                                // notify master that abort has completed by moving to FAILED
+                                notifyFailedSnapshotShard(entry.snapshot(), curr.key, localNodeId, curr.value.reason());
```

Comment:
I'm a little worried that we might be overloading the master by resending this again and again (assume that master is under load processing a lot of pending tasks and we keep resending this on each cluster state update). We'll probably need something along the lines of ShardStateAction.remoteFailedShardsCache`

Reply:
Added cache in https://github.com/elastic/elasticsearch/pull/37686/commits/808c99a1c13caa64578ff351474e150815b81c01